ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/AbortedTxnCleaner.java

```diff
@@ -149,10 +157,20 @@ private void abortCleanUsingAcidDir(CompactionInfo info, String location, long m
 
     boolean success = cleanAndVerifyObsoleteDirectories(info, location, validWriteIdList, table);
     if (success || CompactorUtil.isDynPartAbort(table, info.partName)) {
-      txnHandler.markCleaned(info, false);
+      txnHandler.markCleaned(info, true);
     } else {
       LOG.warn("Leaving aborted entry {} in TXN_COMPONENTS table.", info);
     }
 
   }
+
+  private void handleCleanerAttemptFailure(AbortTxnRequestInfo info) throws MetaException {
+    long defaultRetention = getTimeVar(conf, HIVE_COMPACTOR_CLEANER_RETRY_RETENTION_TIME, TimeUnit.MILLISECONDS);
```

Comment:
should we allow to change this config on the session level? maybe just get it in init method?

Reply:
Done.